packages/iTwinUI-react/src/core/ComboBox/ComboBox.tsx

```diff
@@ -196,21 +236,26 @@ export const ComboBox = <T,>(props: ComboBoxProps<T>) => {
     // When the dropdown opens
     if (isOpen) {
       inputRef.current?.focus(); // Focus the input
-      setFilteredOptions(optionsRef.current); // Reset the filtered list
-      dispatch(['focus']);
+      // Reset the filtered list (does not reset when multiple enabled)
+      !multiple && setFilteredOptions(optionsRef.current);
+      if (!isMultipleEnabled(selected, multiple)) {
+        dispatch({ type: 'focus' });
+      }
```

Comment:
Seems like you could do
`if (!isMultipleEnabled(selected, multiple)) {`
        `dispatch({ type: 'focus' });`
      `} else { setFilteredOptions(optionsRef.current) };`
instead of needing the first line `!multiple && setFilteredOptions(optionsRef.current);`?

Reply:
Both actions are performed when multiple is NOT enabled. Moved both under one if condition.